util/util.go

```diff
@@ -16,3 +18,18 @@ func MakeSignature(size int) ([]byte, error) {
 	b = []byte(base64.StdEncoding.EncodeToString(b))
 	return b, err
 }
+
+// SecretCopy generates a deep copy of a slice containing secrets
+//
+// This function takes a slice of pointers to Secrets and returns a new slice
+// containing deep copies of the original secrets.
+func SecretCopy(secrets []*apiv1.Secret) []*apiv1.Secret {
+	var secretsCopy []*apiv1.Secret
```

Comment:
Wanna go ahead and initialize with length since we know the length?

Reply:
good idea!